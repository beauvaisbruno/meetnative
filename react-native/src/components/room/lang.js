"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const intl_1 = require("../../config/intl");
exports.default = (english, args) => {
    return intl_1.translate(dictionary, english, args);
};
const dictionary = {
    Info: { fr: "Info", de: "Info", it: "Informazioni", es: "Información" },
    yo: { fr: "ans", de: "jä", it: "an", es: "añ" },
    Speak: { fr: "Parle", de: "Sprechen", it: "Parlare", es: "Hablar" },
    Learn: { fr: "Apprend", de: "Lernen", it: "Imparare", es: "Aprender" },
    Message: { fr: "Message", de: "Botschaft", it: "Messaggio", es: "Mensaje" },
    Chat: { fr: "Discussion", de: "Plaudern", it: "Chiacchierare", es: "Charla" }
};
