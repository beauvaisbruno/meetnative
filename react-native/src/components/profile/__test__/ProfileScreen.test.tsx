import { mockMath, unMockMath } from "../../../../jest/testUtils";
mockMath();

import * as React from "react";
import ProfileScreen, { FormUi, getValidationSchema } from "../ProfileScreen";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import { defaultProfileData, initialState } from "../profileReducer";
import { NavigationInjectedProps } from "react-navigation";
import { InjectedFormikProps } from "formik";
import { ValidationError } from "yup";
import { TeachLevel } from "../../../TeachLevel";
import { LearnLevel } from "../../../LearnLevel";
import lang from "../lang";
import { IProfile } from "../../../rootReducers";

const loadProfile = jest.fn();
const updateProfile = jest.fn();
const NavigationAndFormikMock = jest.fn<
  // @ts-ignore
  NavigationInjectedProps & InjectedFormikProps<IProfile, IProfile>
>();

it("ValidationSchema langs", () => {
  let fields = {
    pseudo: "pseudo",
    profession: "profession",
    city: "city",
    age: 21,
    avatar: "avatar",
    lat: 1,
    lon: 1,
    langsToLearn: { key1: "any" },
    langsToTeach: { key2: "any" }
  };
  expect(getValidationSchema().validateSync(fields)).toBe(fields);
  // @ts-ignore
  fields = { ...fields, langsToTeach: { key1: "any" } };
  expect(() => getValidationSchema().validateSync(fields)).toThrowError(
    new ValidationError(
      lang("You couldn't teach and learn {lang}", { lang: "key1" }),
      fields,
      "langsToLearn"
    )
  );
  fields = {
    ...fields,
    // @ts-ignore
    langsToTeach: { key1: TeachLevel.fluent },
    langsToLearn: { key1: LearnLevel.fluent }
  };
  expect(getValidationSchema().validateSync(fields)).toBe(fields);
});

it("ValidationSchema", () => {
  const fieldsSuccess = {
    pseudo: "123",
    profession: "123",
    city: "city",
    age: 21,
    avatar: "avatar",
    lat: 1,
    lon: 1,
    langsToLearn: { key1: "any" },
    langsToTeach: { key2: "any" }
  };
  expect(getValidationSchema().validateSync(fieldsSuccess)).toBe(fieldsSuccess);
  let fields = { ...fieldsSuccess, pseudo: "12" };
  expect(() => getValidationSchema().validateSync(fields)).toThrowError(
    new ValidationError(
      "Pseudo has to be longer than 3 characters!",
      fields,
      "pseudo"
    )
  );
  fields = { ...fieldsSuccess, profession: "12" };
  expect(() => getValidationSchema().validateSync(fields)).toThrowError(
    new ValidationError(
      "Profession has to be longer than 3 characters!",
      fields,
      "profession"
    )
  );
  fields = { ...fieldsSuccess, city: lang("Current location") };
  expect(() => getValidationSchema().validateSync(fields)).toThrowError(
    new ValidationError("Please select a city", fields, "city")
  );
});

it("FormUi", () => {
  const handleSubmit = jest.fn();
  const setFieldTouched = jest.fn();
  const handleChangeImpl = jest.fn(() => null);
  const tree = shallow(
    <FormUi
      {...new NavigationAndFormikMock()}
      {...initialState}
      loadProfile={loadProfile}
      updateProfile={updateProfile}
      values={{ ...defaultProfileData }}
      errors={{}}
      touched={{}}
      handleChange={jest.fn(() => handleChangeImpl)}
      setFieldTouched={setFieldTouched}
      handleSubmit={handleSubmit}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
  const submitBtn = tree.findWhere(node => node.prop("testID") === "updateBtn");
  submitBtn.props().onPress();
  expect(handleSubmit.mock.calls.length).toBe(1);
  tree.setProps({ profileLoad: { fetching: true, error: false } });
  expect(toJson(tree)).toMatchSnapshot();
  tree.setProps({ profileLoad: { fetching: false, error: true } });
  expect(toJson(tree)).toMatchSnapshot();

  tree.setProps({ profileUpdate: { fetching: true, error: false } });
  expect(toJson(tree)).toMatchSnapshot();
  tree.setProps({ profileUpdate: { fetching: false, error: true } });
  expect(toJson(tree)).toMatchSnapshot();

  tree.setProps({
    touched: { pseudo: true },
    errors: { pseudo: "error message" }
  });
  expect(toJson(tree)).toMatchSnapshot();
  tree.setProps({
    touched: { profession: true },
    errors: { profession: "error message" }
  });
  expect(toJson(tree)).toMatchSnapshot();
  tree.setProps({
    touched: { city: true },
    errors: { city: "error message" }
  });
  expect(toJson(tree)).toMatchSnapshot();
  tree.setProps({
    touched: { langsToLearn: true },
    errors: { langsToLearn: "error message" }
  });
  expect(toJson(tree)).toMatchSnapshot();

  const inputs = tree.find("Styled(Input)");
  inputs.forEach((node: any) => {
    node.props().onBlur();
    node.props().onChangeText();
  });
  expect(setFieldTouched.mock.calls.length).toBe(2);
  expect(handleChangeImpl.mock.calls.length).toBe(2);
  unMockMath();
});

it("Formik wrapper", () => {
  const addListener = jest.fn();
  const tree = shallow(
    // @ts-ignore
    <ProfileScreen
      {...{ navigation: { addListener } }}
      {...initialState}
      loadProfile={loadProfile}
      updateProfile={updateProfile}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
  addListener.mock.calls[0][1]();
  expect(loadProfile.mock.calls.length).toBe(1);
  const formikProps = tree.find("Formik").props();
  // @ts-ignore
  formikProps.onSubmit!(undefined);
  expect(updateProfile).not.toBeCalled();
  // @ts-ignore
  formikProps.onSubmit!(defaultProfileData);
  expect(updateProfile.mock.calls[0][0]).toBe(defaultProfileData);
});
