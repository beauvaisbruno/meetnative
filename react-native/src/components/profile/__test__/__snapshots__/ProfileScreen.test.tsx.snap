// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUi 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    />
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 2`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Loading />
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    />
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 3`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Impossible to load the profile
    </Styled(Text)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    />
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 4`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Impossible to load the profile
    </Styled(Text)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    />
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={true}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Component />
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="spinner"
        type="EvilIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 5`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Impossible to load the profile
    </Styled(Text)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    />
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Update impossible
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 6`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Impossible to load the profile
    </Styled(Text)>
    <Styled(Item)
      error={true}
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Text)
      style={
        Object {
          "color": "#d9534f",
        }
      }
    >
      error message
    </Styled(Text)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#d9534f",
          }
        }
      >
        error message
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Update impossible
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 7`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Impossible to load the profile
    </Styled(Text)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      error={true}
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Text)
      style={
        Object {
          "color": "#d9534f",
        }
      }
    >
      error message
    </Styled(Text)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#d9534f",
          }
        }
      >
        error message
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Update impossible
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 8`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Impossible to load the profile
    </Styled(Text)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
      <Styled(Text)
        style={
          Object {
            "color": "#d9534f",
          }
        }
      >
        error message
      </Styled(Text)>
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#d9534f",
          }
        }
      >
        error message
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Update impossible
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormUi 9`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Impossible to load the profile
    </Styled(Text)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Pseudo
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Profession
      </Styled(Label)>
      <Styled(Input)
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        value=""
      />
    </Styled(Item)>
    <Styled(ViewNB)
      style={
        Object {
          "borderBottomColor": "#D9D5DC",
          "borderBottomWidth": 2,
          "marginTop": 5,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#575757",
            "fontSize": 15,
            "marginLeft": 4,
          }
        }
      >
        City
      </Styled(Text)>
      <Component
        city=""
        onChange={[Function]}
      />
    </Styled(ViewNB)>
    <Component
      onChange={[Function]}
      value={21}
    />
    <withNavigation(AvatarSelector)
      onChange={[Function]}
      value="manchots"
    />
    <Component
      onChange={[Function]}
      value={true}
    />
    <Component
      languagesToLearn={
        Object {
          "english": "beginner",
        }
      }
      onChange={[Function]}
    />
    <Component
      languagesToTeach={
        Object {
          "english": "native",
        }
      }
      onChange={[Function]}
    />
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#d9534f",
          }
        }
      >
        error message
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Button)
      block={true}
      disabled={false}
      onPress={[Function]}
      rounded={true}
      success={true}
      testID="updateBtn"
    >
      <Styled(Icon)
        color="white"
        fontSize={30}
        name="cloud-sync"
        type="MaterialCommunityIcons"
      />
      <Styled(Text)>
        Update Profile
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "#d9534f",
        }
      }
    >
      Update impossible
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`Formik wrapper 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)>
    <Formik
      enableReinitialize={true}
      initialValues={
        Object {
          "age": 21,
          "avatar": "manchots",
          "city": "",
          "id": "",
          "isMale": true,
          "langsToLearn": Object {
            "english": "beginner",
          },
          "langsToTeach": Object {
            "english": "native",
          },
          "profession": "",
          "pseudo": "",
        }
      }
      isInitialValid={false}
      onSubmit={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      validationSchema={
        ObjectSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_defaultDefault": [Function],
          "_deps": Array [],
          "_excludedEdges": Array [],
          "_exclusive": Object {},
          "_mutate": false,
          "_nodes": Array [
            "langsToLearn",
            "lon",
            "lat",
            "city",
            "profession",
            "pseudo",
          ],
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "object",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "fields": Object {
            "city": StringSchema {
              "_blacklist": RefSet {
                "list": Set {
                  "Current location",
                },
                "refs": Map {},
              },
              "_blacklistError": [Function],
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
            "langsToLearn": Lazy {
              "_resolve": [Function],
            },
            "lat": NumberSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "number",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
            "lon": NumberSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "number",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
            "profession": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "min": true,
                "required": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
            "pseudo": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "min": true,
                "required": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        }
      }
    >
      <Component />
    </Formik>
  </Styled(ViewNB)>
  <Connect(LogoutButton) />
</Styled(ViewNB)>
`;
